src/forms/DataDrivenForm/formRenderer.js

```diff
@@ -0,0 +1,106 @@
+import React, { Fragment, Component } from 'react';
+import PropTypes from 'prop-types';
+import { Grid, GridItem, Button, Title } from '@patternfly/react-core';
+import { Form, Field } from 'react-final-form';
+import componentMapper from './componentMapper';
+import validationMapper from './validationMapper';
+import { composeValidators } from '../Helpers/helpers';
+
+const renderFields = (properties, uiSchema, globalValidators, options) => Object.keys(properties).map((key) => {
+  const { component, type } = componentMapper(properties[key].type, uiSchema[key] ? uiSchema[key]['ui:widget'] : undefined);
+  const minLength = properties[key].minLength ? validationMapper('minLength')(properties[key].minLength) : undefined;
+  return (
+    <GridItem sm={12} key={key}>
+      <Field
+        name={key}
+        label={properties[key].title}
+        component={component}
+        type={type}
+        validate={composeValidators(globalValidators[key] && globalValidators[key], minLength)}
+        autoFocus={options.autoFocus === key}
+        isRequired={globalValidators[key] && globalValidators[key].name === 'required'}
+      />
+    </GridItem>
+  );
+});
+
+const renderTitle = (title, description) => (
+  <div>
+    {title && <Fragment><Title size="3xl">{title}</Title><hr /></Fragment>}
```

Comment:
Can we omit the `<hr />` tag here? I guess this will be styled by consumers and it will be quite painfull to remove it from DOM by css.